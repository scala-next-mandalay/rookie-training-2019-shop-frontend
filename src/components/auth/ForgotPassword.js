import React from 'react';
import PropTypes from 'prop-types';
import { Box, Container, Link, Button, TextField,Grid}  from '@material-ui/core';
import { withRouter } from 'react-router-dom';
import '../../assets/style.css';
import { makeStyles,createMuiTheme } from '@material-ui/core/styles';
import { ThemeProvider } from '@material-ui/styles';
import { green } from '@material-ui/core/colors';
import CircularProgress from '@material-ui/core/CircularProgress';
 import { FormattedMessage } from 'react-intl';

const useStyles = makeStyles(theme => ({
  btnWrapper: {
    marginTop: theme.spacing(2),
    position: 'relative',
  },
   btnProgress: {
    color: green[500],
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginTop: -12,
    marginLeft: -12,
  },
}));

const ForgotPassword = React.memo(({
  authState, 
  changeAuthState,
  loading,
  error,
  email,
  forgotPassword,
  history
}) => {
  const classes = useStyles();
  const [form, setForm] = React.useState({email:"", password:""});
  
  const handleChangeValue = fieldName => event => {
    const newForm = {...form};
    newForm[fieldName] =  event.target.value;
    setForm(newForm);
  };
  
  const handleSubmit = event => {
    event.preventDefault();
    forgotPassword(form['email']);
  };
  
  const handleSignIn = event => {
    event.preventDefault();
    changeAuthState('signIn');
  };
  const handleClose=event=>{
    event.preventDefault();
    history.push("/");
  };
  const theme = createMuiTheme({
  palette: {
    primary: {
       main:'#cccccc'
    }
   
    
  },
});
  
  const content = (
   <Container component="main" maxWidth="xs" className="container">
  <Box className="authTitle">
  <FormattedMessage id="Label.PasswordReset" defualtMessage="Reset Password" />
  </Box>
  
  
  <Link onClick={handleClose}>
  <span class="close-btn">
  <img src="https://cdn4.iconfinder.com/data/icons/miu/22/circle_close_delete_-128.png" alt="closeicon"></img> 
   </span>
   </Link>

  

  <form onSubmit={handleSubmit}>
      <Box display="flex" mx="auto" my="auto" mt={2} fontWeight={600} color="error.main">
            {error}
      </Box>
        <ThemeProvider theme={theme}>
        <Grid container>
        <Grid item xs={12} sm={12}  >
        <Box mx="auto" my="auto" p={1} mt={5}>
         
            <TextField
              id="email"
              autoComplete="email"
              type="email"
              onChange={handleChangeValue("email")}
              value={form.email}
              label="Email Address"
              variant="outlined"
              required
              fullWidth
              InputLabelProps={{
              style: {
              color: "#ffffff",
          }
        }}
            
            />
            
          
             </Box>
        </Grid>
        </Grid>
         </ThemeProvider>
   
           <Grid container className={classes.btnWrapper}>
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              disabled={loading}
              className="btnSign"
            >
              <FormattedMessage id="Label.SendCode" defualtMessage="Send Code" />
            </Button>
             {loading && <CircularProgress size={24} className={classes.btnProgress} />}
        
            </Grid> 
            
             <Grid container >
             <Grid item xs={12}>
             <Box mt={2}>
              <Link className="ho" variant="body2" onClick={handleSignIn}>
              <FormattedMessage id="Label.BackSignin" defualtMessage="Back to Signin" />
            </Link>
            </Box>
            </Grid>
            </Grid>
            
     </form>
    </Container>
  );
  return (authState === 'forgotPassword') ? content : null;
});

ForgotPassword.propTypes = {
  authState: PropTypes.string.isRequired,
  changeAuthState: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  email: PropTypes.string,
  forgotPassword: PropTypes.func.isRequired,
};

export default withRouter(ForgotPassword);