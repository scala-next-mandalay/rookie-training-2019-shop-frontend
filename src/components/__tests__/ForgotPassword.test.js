import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import ForgotPassword from '../auth/ForgotPassword';
import Parent from '../Parent';

/*global jest*/
/*global expect*/
/*global MouseEvent*/
/*global Event*/

let container;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
});



const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
let authState = "forgotPassword";
const changeAuthState = jest.fn((signIn) => {
  authState = signIn;
});

const loading = false;
const error = "Forgot password ";
let email = "";
const forgotPassword = jest.fn((email) => {
  email = email;
});

describe("ForgotPassword component snapshot", () => {
  it('matches the snapshot', () => {
    const ForgotPasswordSnapshot = renderer.create(
    <Parent>
      <ForgotPassword
        authState={authState}
        changeAuthState={changeAuthState}
        loading={loading}
        error={error}
        email={email}
        forgotPassword={forgotPassword}
      />
     </Parent>

    ).toJSON();
    expect(ForgotPasswordSnapshot).toMatchSnapshot();

  });

})


describe("Testing ForgotPassword", () => {
  it('Testing textfied and  Send Code button with no data', () => {
    act(() => {
      ReactDOM.render(  
          <Parent>
        <ForgotPassword
        authState={authState}
        changeAuthState={changeAuthState}
        loading={loading}
        error={error}
        email={email}
        forgotPassword={forgotPassword}
      /></Parent>, container);
    });
    
    //With no input data
    const inputArr = document.querySelectorAll('input');
     act(() => {
      valueSetter.call(inputArr[0], '');
      inputArr[0].dispatchEvent(new Event('change', { bubbles: true }));
    });
    
    const buttonArr1 = document.querySelectorAll('button');
    act(() => {
      buttonArr1[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));

    });
     expect(forgotPassword).toHaveBeenCalledWith("");
     
     //With  input data 
    const inputArr1 = document.querySelectorAll('input');
    act(() => {
      valueSetter.call(inputArr1[0], 'a@gmail.com');
      inputArr1[0].dispatchEvent(new Event('change', { bubbles: true }));
    });
    expect(inputArr1[0].value).toBe('a@gmail.com');
    
    const buttonArr = document.querySelectorAll('button');
    act(() => {
      buttonArr[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(forgotPassword).toHaveBeenCalledWith("a@gmail.com");
    
    const linkArr = document.querySelectorAll('a');
    act(() => {
      linkArr[1].dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(changeAuthState).toHaveBeenCalledWith("signIn");
  
  })

})