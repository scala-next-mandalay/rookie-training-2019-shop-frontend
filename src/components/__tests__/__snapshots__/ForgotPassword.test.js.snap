// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword component snapshot matches the snapshot 1`] = `
<main
  className="MuiContainer-root container MuiContainer-maxWidthXs"
>
  <div
    className="MuiBox-root MuiBox-root-11 authTitle"
  >
    <span>
      Reset Password
    </span>
  </div>
  <a
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    <span
      className="close-btn"
    >
      <img
        alt="closeicon"
        src="https://cdn4.iconfinder.com/data/icons/miu/22/circle_close_delete_-128.png"
      />
    </span>
  </a>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-48"
    >
      Forgot password 
    </div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
      >
        <div
          className="MuiBox-root MuiBox-root-153"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="email"
              style={
                Object {
                  "color": "#ffffff",
                }
              }
            >
              Email Address
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                â€‰
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="email"
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="email"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-211 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-212"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-btnWrapper-1 MuiGrid-container"
    >
      <button
        className="MuiButtonBase-root MuiButton-root btnSign MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          <span>
            Send Code
          </span>
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiBox-root MuiBox-root-336"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover ho MuiTypography-body2 MuiTypography-colorPrimary"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            <span>
              Back to Signin 
            </span>
          </a>
        </div>
      </div>
    </div>
  </form>
</main>
`;
